### PR TITLE
issue-06 - Implement-database-connection

### PR DESCRIPTION
**Pull request by Adriano-Santtos**
Below are the most important changes:

### Build and Configuration:

* [`.github/workflows/build.yml`](diffhunk://#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721R1-R39): Added a new GitHub Actions workflow to build the Spring application, set up JDK 21, cache Gradle dependencies, and run tests.
* [`src/main/resources/application.yaml`](diffhunk://#diff-b4c0421bcb3280a8e738550c830dba197057c515bcdd9579a52cbfe9adca2061R1-R35): Added configuration for Spring Boot, including JPA settings, datasource details, Flyway migration paths, and management endpoints.

### Documentation:

* [`README.md`](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L1-R21): Updated the README to describe the purpose of the PoC, the technologies used, and the workflow followed.

### Codebase Enhancements:

* [`src/main/kotlin/the/coding/force/exploring_kotlin_coroutines/entities/DataEntity.kt`](diffhunk://#diff-5736347792e1d063abdd23d706eb51793c9f66a87c331b350d42980f6887eaa6R1-R22): Added a new `DataEntity` class to represent data in the database.
* [`src/main/kotlin/the/coding/force/exploring_kotlin_coroutines/repository/DataRepository.kt`](diffhunk://#diff-09f2b2f9783d594617cb0a28e412f1730c0a0d13689345b6175014ae6b3367bbR1-R6): Added a new `DataRepository` interface extending `JpaRepository` for `DataEntity`.

### Removed Configuration:

* [`src/main/resources/application.properties`](diffhunk://#diff-54eeffbae371fcd1398d4ca5e89a1b8118208b7bb2f8ddf55c1aa2f7d98ab136L1-L3): Removed the old Spring application properties file.

### Evidences
* Database successfully conncected locally:
image

### CODE REVIEW

File: src/main/resources/application.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    show-sql: 'false'
+    properties:
+      hibernate:
+        format_sql: 'true'
+    database-platform: org.hibernate.dialect.PostgreSQLDialect
+  datasource:
+    password: local
+    driver-class-name: org.postgresql.Driver
+    username: local
+    url: jdbc:postgresql://localhost:5432/task-manager-poc
```
Comment (Adriano-Santtos) [inline src/main/resources/application.yaml:18]: wrong name

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    container_name: grafana
+    ports:
+      - "3000:3000"
+    restart: unless-stopped
+    environment:
+      - GF_SECURITY_ADMIN_USER=admin
+      - GF_SECURITY_ADMIN_PASSWORD=grafana
+    volumes:
+      - ./grafana/provisioning:/etc/grafana/provisioning
+    networks:
```
Comment (joseiedo) [inline docker-compose.yml:27]:
```suggestion
      - GF_SECURITY_ADMIN_PASSWORD=grafana
    volumes:
      - ./docker/grafana/provisioning:/etc/grafana/provisioning
    networks:
```

Faltou corrigir o path aqui pro grafana subir já configurado. Também dá pra mover tudo que tá na pasta docker pra raiz. Tu que sabe!

**Review by joseiedo — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 2 above)
Comment (Adriano-Santtos) [reply docker-compose.yml]: Boa mestre, vou mover tudo para dentro de docker
Movi para fora para fazer um teste e esqueci de mover novamente

**Review by joseiedo — APPROVED** [review]

**Review by Lucass-trra — COMMENTED** [review]
Código bem estruturado, a ideia dos scripts SQL foi interessante para que ja tenha um esquema de tabela definido para que já se possa popular os primeiros dados, sem erros vigentes.

**Pull request merged**